addons/adyen_platforms/models/adyen_account.py

```diff
@@ -213,75 +284,91 @@ def action_create_redirect(self):
+    def _format_data(self, values):
+        fields = set(values.keys())
         data = {
-            'accountHolderCode': self.account_holder_code,
-            'accountHolderDetails': {
-                'address': {
-                    'country': self.country_id.code,
-                    'stateOrProvince': self.state_id.code or None,
-                    'city': self.city,
-                    'postalCode': self.zip,
-                    'street': self.street,
-                    'houseNumberOrName': self.house_number_or_name,
-                },
-                'email': self.email,
-                'fullPhoneNumber': self.phone_number,
-            },
-            'legalEntity': 'Business' if self.is_business else 'Individual',
+            'accountHolderCode': values.get('account_holder_code') or self.account_holder_code,
         }
 
-        if self.is_business:
-            data['accountHolderDetails']['businessDetails'] = {
-                'legalBusinessName': self.legal_business_name,
-                'doingBusinessAs': self.doing_business_as,
-                'registrationNumber': self.registration_number,
-            }
-        else:
-            data['accountHolderDetails']['individualDetails'] = {
-                'name': {
-                    'firstName': self.first_name,
-                    'lastName': self.last_name,
-                    'gender': 'UNKNOWN',
-                },
-                'personalData': {
-                    'dateOfBirth': str(self.date_of_birth),
-                }
+        # *ALL* the address fields are required if one of them changes
+        address_fields = {'country_id', 'state_id', 'city', 'zip', 'street', 'house_number_or_name'}
+        if address_fields & fields:
+            data.setdefault('accountHolderDetails', {})
+            country_id = self.env['res.country'].browse(values.get('country_id')) if values.get('country_id') else self.country_id
+            state_id = self.env['res.country.state'].browse(values.get('state_id')) if values.get('state_id') else self.state_id
+            data['accountHolderDetails']['address'] = {
+                'country': country_id.code,
+                'stateOrProvince': state_id.code or None,
+                'city': values.get('city') or self.city,
+                'postalCode': values.get('zip') or self.zip,
+                'street': values.get('street') or self.street,
+                'houseNumberOrName': values.get('house_number_or_name') or self.house_number_or_name,
             }
 
-            # documentData cannot be present in the data if not set
-            if self.document_number:
-                data['accountHolderDetails']['individualDetails']['personalData']['documentData'] = [{
-                    'number': self.document_number,
-                    'type': self.document_type,
-                }]
+        if 'email' in fields:
+            data.setdefault('accountHolderDetails', {}).update({'email': values.get('email')})
```

Comment:
You've checked that the entry is in the dict, just use it. And using `update` with a literal dict is also unnecessary.
```suggestion
        if 'email' in fields:
            details['email'] = values['email']
```

Reply:
Also even if they're the same data, checking that the entry exists in `fields` then getting it from `values` is confusing. Just check that `'email' in values`.